micropip/_commands/check_package_dependencies.py

```diff
@@ -0,0 +1,105 @@
+import importlib.metadata
+import json
+
+from packaging.requirements import Requirement
+from packaging.utils import canonicalize_name
+
+from .._compat import REPODATA_PACKAGES
+
+
+def check_package_dependencies(
+    package_name, *, extras=None, fix_deps=False, recursive=True
+):
+    """Check and optionally fix the list of dependencies for this package
+
+    If you have manually installed a package and dependencies from wheels,
+    the dependencies will not be correctly setup in the package list
+    or the pyodide lockfile generated by freezing. This method checks
+    if the dependencies are correctly set in the package list, and if
+    fix_deps is True, it will add missing dependencies.
+
+    Parameters
+    ----------
+    package_name (string):
+        The name of the package to check.
+
+    extras (list):
+        List of extras for this package.
+
+    fix_deps (boolean):
+        If this is True, any missing dependencies that are in
+        the current package list will be added to the current
+        package.
+
+    recursive (boolean):
+        If this is True, dependencies of dependencies will be checked.
```

Comment:
I'm not that big of a fan of these boolean arguments, especially since we don't use them. If possible it's better to make multiple different functions and compose them to get different behaviors rather than taking arguments.